Components/Form.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+
+function Form() {
+  const [todo, setTodo] = useState("");
+  const [todolist, setTodoList] = useState([]);
+  const [isChecked, setChecked] = useState(false);
+
+  const hanldeChange = (e) => {
+    setTodo(e.target.value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  const addTodo = () => {
+    //console.log(todo); test
+    if (todo !== "") {
+      const todoInsert = {
+        id: Math.floor(Math.random() * 10000), //Generate random id from 0 to 10000
+        value: todo,
+        isCompleted: false,
+      };
+
+      setTodoList([...todolist, todoInsert]); //... operator "spreads out the own enumerable properties as properties of the element you're creating" stack overflow
+    }
+  };
+
+  const deleteTodo = (e, id) => {
+    e.preventDefault();
+    setTodoList(todolist.filter((t) => t.id != id));
+  };
+
+  const completeTodo = (e) => {
+    e.preventDefault();
+    setChecked(e.currentTarget.checked);
+  };
+
+  return (
+    <form action="" onSubmit={handleSubmit}>
+      <label htmlFor="newItem">Nuovo item</label>
+      <input
+        type="text"
+        name="newItem"
+        id="newItem"
+        onChange={(e) => hanldeChange(e)}
+        placeholder="Insert todo"
+      />
+      <button onClick={addTodo}>Add</button>
+
+      {todolist !== [] ? (
```

Comment:
This comparison is not doing anything. As most things is javascript are objects (arrays are also objects), all this comparison is doing is checking their memory reference.
Open google chrome console and type `[] !== []`, you will see that you'll get `true` as a result, as newly instantiated objects are always different (they have different memory addresses).

<img width="365" alt="Schermata 2021-03-27 alle 17 48 32" src="https://user-images.githubusercontent.com/4640135/112727840-a7199b80-8f24-11eb-81ca-193313688c26.png">


If what you want to do here is check whether the array is empty or not, simply check its length (e.g. `todoList.length > 0`